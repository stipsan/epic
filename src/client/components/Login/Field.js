import { PropTypes } from 'react'

import cx from './style.scss'

const Field = ({
  input,
  meta,
  ...custom,
  asyncValidating,
  touched,
  error,
  submitting,
  className,
 ...props
}) => (
  <p
    className={cx('control', `control-${custom.name}`, className, {
      error: meta.touched && meta.error,
      'is-loading': meta.asyncValidating,
    })}
  >
    <input
      {...input}
      className={cx(`input-${input.props.name}`, {
        'is-danger': touched && error,
      })}
      autoComplete="email"
      disabled={meta.submitting}
    />
    {meta.touched && meta.error &&
      <span className={cx('help', 'is-danger')}>{meta.error}</span>
    }
  </p>
)

Field.propTypes = {
  asyncValidating: PropTypes.bool,
  className: PropTypes.string,
  error: PropTypes.string,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  submitting: PropTypes.bool,
  touched: PropTypes.bool,
  type: PropTypes.string,
}

export default Field
