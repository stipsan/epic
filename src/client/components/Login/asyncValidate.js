import {
  CHECK_EMAIL_EXISTS_ASYNC,
} from '../../constants/ActionTypes'

const asyncValidate = (values, dispatch) => new Promise((resolve, reject) =>
  dispatch({ type: CHECK_EMAIL_EXISTS_ASYNC, payload: { values, resolve, reject } })
)

export default asyncValidate
