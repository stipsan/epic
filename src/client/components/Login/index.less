.uk-button.uk-form-icon [class*='uk-icon-'] {
  color: inherit;
  right: 0px;
}
.uk-width-1-1 > .uk-form-icon {
  width: 100%;
}
