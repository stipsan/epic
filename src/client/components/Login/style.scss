@import '../../style/_config';

.section {
  composes: section from '../../style/_section.scss';
  
  align-items: center;
  justify-content: center;

  background: $bg-login;
  
  form {
      background: hsla(0, 0%, 0%, .1);
      display: flex;
      overflow: hidden;
      max-width: 500px;
      transition: background 150ms ease, opacity 300ms ease-out;
      width: 100%;
      
      input {
          background: hsla(0, 0%, 100%, 0);
          color: hsla(0, 0%, 0%, .9);
          padding: 0.5rem 1rem;
          width: 100%;
          
          &::placeholder {
              color: hsla(0, 0%, 0%, .4);
          }
      }
      button {
          background: darken($bg-login, 40%);
          color: $bg-login;
          opacity: 1;
          padding: 0.5rem .1rem;
          transition: all 300ms cubic-bezier(0.65, 0.05, 0.36, 1);
          transform: translateX(0%);
          text-transform: uppercase;
          width: 5rem;
          
          &:hover {
              color: lighten($bg-login, 30%);
              background: darken($bg-login, 30%);
          }
          &:active {
              color: lighten($bg-login, 40%);
              background: darken($bg-login, 40%);
          }
      }
      
      input, button {
          border: none;
          font-size: 1rem;
          margin: 0;
          
          &:focus {
              outline: 0;
          }
      }
      
      &:hover, &:focus, &:active {
          background: hsla(0, 0%, 100%, .1);
      }
      
      &:invalid, input:invalid + {
          button {
              opacity: 0;
              transform: translateX(100%);
          }
      }
  }
}
.pendingMessage {
  transition: opacity 300ms ease-out;
}

.container {
  composes: container from '../../style/bulma.scss';
}
.hero {
  composes: hero is-dark is-bold is-fullheight from '../../style/bulma.scss';
}
.hero-content {
  composes: hero-content from '../../style/bulma.scss';
}
.hero-header {
  composes: hero-header from '../../style/bulma.scss';
  
  text-align: center;
  margin-top: 10px;
}
.hero-footer {
  composes: hero-footer from '../../style/bulma.scss';
}
.tabs {
  composes: tabs is-boxed is-fullwidth from '../../style/bulma.scss';
}

// @TODO http://codepen.io/WebSonick/pen/vjmgu/

// @TODO http://codepen.io/kpk/details/yGLnw