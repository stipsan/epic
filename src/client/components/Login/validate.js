const validate = values => {
  const errors = {}
  console.log('validator', values)
  if (!values.has('email')) {
    errors.email = 'Required'
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.get('email'))) {
    errors.email = 'Invalid email address'
  }
  if (!values.has('username')) {
    errors.username = 'Required'
  }
  return errors
}

export default validate
