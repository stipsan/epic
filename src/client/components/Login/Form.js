import { Component, PropTypes } from 'react'

import CheckEmail from './steps/CheckEmail'
import Login from './steps/Login'
import Register from './steps/Register'

class Form extends Component {

  static propTypes = {
    handleCheckEmail: PropTypes.func.isRequired
  }

  state = {
    page: 1
  }

  nextPage = () => this.setState({ page: this.state.page + 1 })

  previousPage = () => this.setState({ page: this.state.page - 1 })

  render() {
    const { onSubmit } = this.props
    const { page } = this.state
    return <div>
      {1 === page && <CheckEmail onSubmit={this.nextPage} />}
      {2 === page && <Login previousPage={this.previousPage} onSubmit={this.nextPage} />}
      {3 === page && <Register previousPage={this.previousPage} onSubmit={onSubmit} />}
    </div>

  }
}

export default Form
