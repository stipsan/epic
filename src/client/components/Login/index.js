import Field from 'epic-client/components/Form/Field'
import { Component, PropTypes } from 'react'
import { Link } from 'react-router'

import FieldComponent from './Field'

const formStyle = Object.freeze({
  width: '250px'
})

export default class Login extends Component {
  static propTypes = {
    checkIfEmailExists: PropTypes.func.isRequired,
    createUserWithEmailAndPassword: PropTypes.func.isRequired,
    doesEmailExist: PropTypes.bool,
    isRequestPending: PropTypes.bool.isRequired,
    signInWithEmailAndPassword: PropTypes.func.isRequired,
    submitting: PropTypes.bool,
  }

  render() {
    const {
      checkIfEmailExists,
      createUserWithEmailAndPassword,
      signInWithEmailAndPassword,
      doesEmailExist,
      submitting,
    } = this.props

    return (
      <div className="uk-flex uk-flex-column uk-height-1-1">
        <div className="uk-flex-item-auto uk-flex uk-flex-center uk-flex-middle">
          <div className="uk-text-center" style={formStyle}>
            <img
              src="/favicons/icon.svg"
              className="uk-margin-bottom"
              width="64"
              height="64"
              role="presentation"
            />
            <form className="uk-panel uk-panel-box uk-form">
              <Field
                name="email"
                type="email"
                placeholder="E-mail"
                submitting={submitting}
                component={FieldComponent}
              />
              <button>Submit</button>
            </form>
          </div>
        </div>
        <div className="uk-flex-item-none">
          <ul className="uk-subnav uk-subnav-line uk-flex-center">
            <li><Link to="/about">{'About'}</Link></li>
          </ul>
        </div>
      </div>
    )
  }
}
