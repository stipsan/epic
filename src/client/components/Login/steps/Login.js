import { PropTypes } from 'react'
import { Link } from 'react-router'
import { Field, reduxForm } from 'redux-form/immutable'

import asyncValidate from '../asyncValidate'
import cx from '../style.scss'
import validate from '../validate'
import FieldComponent from '../Field'

const CheckEmailForm = props => {
  const { handleSubmit, submitting } = props
  console.log('props', props)
  return (
    <form onSubmit={handleSubmit} className={cx('form')}>
      <Field
        name="email"
        type="email"
        placeholder="E-mail"
        autoComplete="email"
        className={cx('is-marginless', 'no-bottom-rounded-border')}
        submitting={submitting}
        component={FieldComponent}
      />
      <Field
        name="password"
        type="password"
        placeholder="Password"
        autoComplete="password"
        className="no-top-rounded-border"
        submitting={submitting}
        autoFocus
        component={FieldComponent}
      />
      <p className={cx('control')}>
        <button
          className={cx('login-button', { 'is-loading': submitting })}
          disabled={submitting}
        >{'Log in'}</button>
      </p>
      <p className={cx('control')}>
        <Link to="/forgot" className={cx('forgot-password')}>{'Forgot your password?'}</Link>
      </p>
    </form>
  )
}

CheckEmailForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
}

export default reduxForm({
  form: 'login',
  destroyOnUnmount: false,
  validate,
  asyncValidate,
  asyncBlurFields: ['email'],
})(CheckEmailForm)
