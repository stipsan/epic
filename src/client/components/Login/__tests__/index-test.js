import renderer from 'react-test-renderer'

import Login from '../index'

describe('<Login />', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <Login />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
