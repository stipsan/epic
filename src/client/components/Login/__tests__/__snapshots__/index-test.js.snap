exports[`<Login /> should render correctly 1`] = `
<div
  className="uk-flex uk-height-1-1">
  <div
    className="uk-vertical-align-middle">
    <div
      className="hero-head">
      <div
        className="container">
        <img
          height="64"
          role="presentation"
          src="/favicons/icon.svg"
          width="64" />
      </div>
    </div>
    <div
      className="hero-body">
      <div
        className="container">
        <div />
      </div>
    </div>
    <div
      className="hero-footer">
      <nav
        className="footer-tabs">
        <div
          className="container">
          <ul>
            <li>
              <a
                onClick={[Function bound handleClick]}
                style={Object {}}>
                About
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
